### PR TITLE
feat(secrets): add V2 evidence contracts and release claim gate

### PR DESCRIPTION
**Pull request by kantorcodes**
## Summary

- adds machine-readable V2 source-capability, reason-code, product-boundary, and capability-evidence contracts
- adds the V2 contract loader and validation tests
- adds the release-claim gate and its tests
- updates the test-suite ratchet only for the added coverage

## Release safety

This PR does not make or authorize a public Snyk-parity claim. The release claim remains denied unless the evidence gate passes against complete, current proof inputs. It targets `release/3.0` only and must not be merged into `main` as part of this work.

## Validation

Exact-head CI, security checks, and review must pass before squash merge.

## Summary by CodeRabbit

- **New Features**
  - Added comprehensive v2 contracts for secrets scanning coverage, detection outcomes, ignore rules, custom rules, capabilities, metrics, and release evidence.
  - Added support for clearly defined plan capabilities, repository limits, routing controls, governance rules, and local-operation guarantees.
  - Added structured reason codes and source-capability coverage across files, archives, Git, editors, operating systems, validators, and notifications.

- **Validation**
  - Added release-readiness validation for capability evidence, exact release commits, required evidence, supported states, and labeled gaps.
  - Strengthened validation to reject unknown fields, invalid data, duplicate entries, and prohibited sensitive content.

### CODE REVIEW

**Comment by kantorcodes** [comment]
@coderabbitai review

@kilocode-bot review this exact head. Validate the evidence contract, fail-closed release-claim gate, privacy boundaries, schema consistency, path safety, deterministic behavior, and tests. Report any actionable issue before merge.

**Comment by kantorcodes** [comment]
@kilocode-bot fix the exact-head Ruff failures only: import `Mapping` and `Sequence` from `collections.abc` in `src/codex_plugin_scanner/guard/secrets/contracts_v2.py`, keep `Final` and `cast` from `typing`, and remove the unnecessary quotes around the `SecretScanCoverageV2` return annotation. Do not change contract semantics or broaden the PR.

**Comment by kantorcodes** [comment]
@kilocode-bot address every still-valid CodeRabbit review item on the exact current head, keeping this PR strictly scoped to HGS2-T016 through HGS2-T030 and targeting release/3.0 only.

Required changes:
1. In `SecretScanCoverageV2`, move the `truncation_codes => partial` and `completed_refs ⊆ requested_refs` invariants into `__post_init__` so direct construction and `from_mapping` share enforcement; remove duplicate post-construction checks.
2. Add one module-level `_COMMIT_SHA` compiled regex and reuse it for every exact commit validation.
3. Keep `validate_capability_manifest` as the canonical parity implementation. Add a strict helper that parses the capability-evidence manifest mapping into `CapabilityEvidenceV2` rows, including allowed-key validation and gap-label handling needed by the release gate. Refactor `guard_secrets_release_claim_gate.py` to derive states from `ParityState`, consume and validate the manifest's `parity_states` and `claim_policy`, and delegate release-parity validation to the canonical contract implementation. Preserve input-error exit code 2, validation-error exit code 1, success 0, and fail closed.
4. Send all gate diagnostics to stderr.
5. Update `guard-secrets-product-boundaries.v2.json` so each plan has a consistent typed shape: numeric `private_repositories`, a separate `private_repositories_mode`, boolean `governed_ignores`, and `governed_ignores_enforced`. Preserve the intended Free/Solo/Pro/Team commercial boundary.
6. Add tests for direct-construction coverage invariants, malformed exact release SHA, duplicate capability IDs, missing required capability, `OUTCOME_SURFACE_MAPPING` exhaustiveness, gate `main()` exit codes 0/1/2, parity-state/claim-policy drift, and reason-code registry equality with the JSON contract.
7. Fold the redundant `tests/test_guard_secret_release_claim_gate_inputs.py` coverage into the canonical gate test file, then delete the redundant file.
8. Update the test-suite ratchet only to the exact new collected-case and source-line counts reported by the repository ratchet.
9. Run Ruff format/check on changed Python, focused tests, and the repository-required exact-head gates. Do not add temporary workflows, bypasses, allowlists, raw candidate output, or competitor strategy.

After the implementation, request a fresh CodeRabbit review and leave the branch ready for exact-head CI and merge.

**Comment by kantorcodes** [comment]
@kilocode-bot also update this branch onto the current `release/3.0` head before final validation so every check and review runs against the exact latest target branch, resolving any ratchet drift without dropping unrelated release changes.

**Comment by kantorcodes** [comment]
@coderabbitai review

File: scripts/ci/guard_secrets_release_claim_gate.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                )
+        elif not isinstance(gap_label, str) or not gap_label.strip():
+            errors.append(
+                f"{capability_id}: non-release state requires an explicit gap label"
+            )
+        if require_parity and capability_id in required_capabilities:
+            if state not in _RELEASE_STATES:
+                errors.append(f"{capability_id}: not release-candidate verified")
+            elif release_commit != exact_release_commit:
+                errors.append(f"{capability_id}: evidence is bound to another commit")
```
[the remark this answers is not in the document]
Comment (kantorcodes) [reply scripts/ci/guard_secrets_release_claim_gate.py]: @coderabbitai apply the remaining review suggestions

[the remark this answers is not in the document]
Comment (kantorcodes) [reply scripts/ci/guard_secrets_release_claim_gate.py]: @coderabbitai apply all seven remaining nitpick suggestions and commit the fixes to this branch
